talisker/gunicorn.py

```diff
@@ -112,15 +113,16 @@ def gunicorn_child_exit(server, worker):
         server.SIG_QUEUE.append('SIGCUSTOM')
 
 
-def gunicorn_pre_request(worker, req):
-    """Gunicorn pre_request hook.
-
-    Clear any previous contexts on new request.
-
-    Note: we do this on way in, rather than the way out, to preserve the
-    request_id context when there's a timeout.
-    """
-    talisker.clear_contexts()
+def gunicorn_worker_exit(server, worker):
+    """Logs any requests that are still in flight."""
+    now = time.time()
+    for id, env in talisker.wsgi.REQUESTS.items():
```

Comment:
drop `id` since it's not used, and iterate over `.values()`
```suggestion
    for env in talisker.wsgi.REQUESTS.values():
```